### PR TITLE
FABN-1539 NodeSDK limit "all" transaction event listening

### PR DESCRIPTION
**Pull request by harrisob**
Only notify tranaction event listeners of 'ENDORSER_TRANSACTION'
and not include other types like config changes. V2 lifecycle
changes are included.

Update the TS scenario tests with testing from the js scenario
testing so the js scenario may be removed while adding the new base
event testing of events.

Signed-off-by: Bret Harrison <beharrison@nc.rr.com>

### CODE REVIEW

File: test/ts-scenario/features/base_api_v2.feature [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	And I regisister a block listener named myBlockListener with myEventService2 for startBlock 1 and endBlock 2 as client fred
+	And I regisister a chaincode listener named myChaincodeListener with myEventService2 for createCar event on contract fabcar as client fred
+	And I regisister a transaction listener named myTransactionListener with myEventService2 for all transactions as client fred
+	And I start the event service myEventService2 as full blocks to start at block 0 and end at block 7 as client fred
+	And I build a new endorsement request named replay4 for smart contract named fabcar with arguments [createCar,2005,Ford,Taurus,white,Henry] as client fred on channel eventschannel
+	And I commit the endorsement request named replay4 as client fred on channel eventschannel
+	Then the event listener myBlockListener of myEventService2 has results matching {"block":"2"} as client fred
+	Then the event listener myChaincodeListener of myEventService2 has results matching {"createCar":"PTCurser"} as client fred
+	Then the event listener myTransactionListener of myEventService2 has results matching {"transaction":"6"} as client fred
```
Comment (bestbeforetoday) [inline test/ts-scenario/features/base_api_v2.feature:66]: Is this relying on other scenarios having run (and the order in which they run), including ones in feature files outside of this one? Expecting blocks 0 to 7 to exist and contain 6 transactions. If so, we might hit problems down the line if either tests change, just a subset of tests are run, or tests run in an unexpected order. Is there any practical alternative?

**Review by bestbeforetoday — COMMENTED** [review]
Great to remove all those old / redundant test files! A few comments / questions but I think only two raised some concern for me:
- Whether `endBlock` was guaranteed to have a suitable value at that point in the EventService code.
- Whether there was interdependency between scenario tests that might cause problems.

File: fabric-common/types/index.d.ts [hunk 2]
```diff
@@ -239,19 +239,19 @@ export interface ChaincodeEvent {
...
+	readonly endBlock?: Long | string;
+	readonly blockType: BlockType;
 	constructor(chaincodeName: string, channel: Channel);
 	public setEventer(discoverer: Eventer): EventService;
 	public getLastBlockNumber(): Long;
 	public close(): void;
-	public build(idContext: IdentityContext, request: any): Buffer;
-	public send(request: StartRequestOptions): Promise<any>;
+	public build(idContext: IdentityContext, request: StartRequestOptions): Buffer;
+	public send(request: SendEventOptions): Promise<any>;
```
[the remark this answers is not in the document]
Comment (bestbeforetoday) [reply fabric-common/types/index.d.ts:250]: This code change happened in another PR anyway. Not sure about the JSDoc.

File: fabric-network/src/impl/event/eventservicemanager.ts [hunk 3]
```diff
@@ -60,7 +60,7 @@ export class EventServiceManager {
 		eventService.build(this.identityContext, options);
 		eventService.sign(this.identityContext);
 		// targets must be previously assigned
-		await eventService.send(options);
+		await eventService.send({});
```
[the remark this answers is not in the document]
Comment (bestbeforetoday) [reply fabric-network/src/impl/event/eventservicemanager.ts:63]: Another PR already changed this

File: fabric-common/lib/EventService.js [hunk 4]
```diff
@@ -488,10 +488,24 @@ class EventService extends ServiceAction {
 							logger.debug('on.data - status received when newest block seen: %s block_num: %s',
 								deliverResponse.status, this.lastBlockNumber.toNumber());
 							this._close(new Error(`Newest block received:${this.lastBlockNumber.toNumber()} status:${deliverResponse.status}`));
-						} else if (this.endBlock) {
-							logger.debug('on.data - status received before the endblock has been seen');
-							this._close(new Error(`End block of ${this.endBlock.toNumber()}` +
+						} else if (this.endBlock.greaterThan(this.lastBlockNumber)) {
```
[the remark this answers is not in the document]
Comment (harrisob) [reply fabric-common/lib/EventService.js:491]: fixed, thanks

File: fabric-common/lib/EventService.js [hunk 5]
```diff
@@ -982,19 +996,23 @@ class EventService extends ServiceAction {
...
 			}
 		} else {
 			logger.debug('%s full block number=%s', method, full_block.header.number);
 			const txStatusCodes = full_block.metadata.metadata[fabprotos.common.BlockMetadataIndex.TRANSACTIONS_FILTER];
 			for (let index = 0; index < full_block.data.data.length; index++) {
 				const channel_header = full_block.data.data[index].payload.header.channel_header;
-				this._callTransactionListener(channel_header.tx_id,
-					txStatusCodes[index],
-					full_block.header.number);
+				if (channel_header.type === 3) {
```
[the remark this answers is not in the document]
Comment (harrisob) [reply fabric-common/lib/EventService.js:1011]: changed

File: test/ts-scenario/features/base_api_v2.feature (hunk 1 above)
Comment (harrisob) [reply test/ts-scenario/features/base_api_v2.feature:66]: these should be in the same scenario

**Comment by harrisob** [comment]
I have another change comming

**Review by bestbeforetoday — APPROVED** [review]

**Pull request merged**